packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1470,36 +1472,87 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         return this.context.submitSignalFn(envelope);
     }
 
+    /**
+     * Updates the garbage collection nodes of this node's children:
+     * - Prefixs the child's id to the id of each node returned by the child.
+     * @param chilGCNodes - The child's garabage collection nodes.
+     * @param childId - The id of the child node.
+     * @returns the updated GC nodes of the child.
+     */
+    private updateChildGCNodes(chilGCNodes: IGraphNode[], childId: string): IGraphNode[] {
```

Comment:
I'm not sure what value does this function add on top of `normalizeAndPrefixGCNodeIds`?  Seems like this could be removed altogether and callers can just call `normalizeAndPrefixGCNodeIds`.

Reply:
There are 2 reasons I did it (although not strong reasons) -
1. I might add a reference back to container runtime to all its child nodes. I have not finalized it yet.
2. I want to keep the implementation in dataStoreRuntime and containerRuntime as similar as possible. From GC's perspective, they are the same.

I will remove this method if it doesn't make sense once I have the main GC implementation.

Reply:
I guess that makes sense. I feel like that code doesn't need to live here though. Can you just have a function that calls both `normalizeAndPrefixGCNodeIds` _and_ `addRouteToAllGCNodes` that takes in the absolutePath as a parameter?

Just feel like I wanna keep these classes as slim as possible (though we are already _not_ succeeding with ContainerRuntime).

Reply:
Yes, I agree that we want to keep ContainerRuntime and DataStoreRuntime as slim as possible. But this is not getting added to the IContainerRuntime interface so it shouldn't be a big deal.
Also, I think garbage collection will be core to every layer we have so we will have some GC related functionality.

Maybe the long term solution is to move garbage collection functionality to a separate helper class so the core classes are cleaner.

Reply:
I guess I just mean these `updateChildGCNodes` functions are all roughly the same, which just call the 2 external helper functions. Couldn't that itself be a helper function?

Reply:
Also, not a big deal if it stays here, just feels a little unnecessary and slightly code duplication

Reply:
I guess you are right. If every node is going to do the same thing, it might be worth to add it as a helper. Tell you what, once I have the GC algorithm in place, it will be much more clearer what each node looks like and I can handle it then.
Right now, I am not sure if containerRuntime will need to call both these helpers and if not, I will remove the updateChildGCNodes method.

Reply:
Sounds totally reasonable, I like it! My view is that this is a "big" PR which gets everything working (as needed for subsequent PRs) and then we can do small, incremental PRs with little-to-no functional change once we have a clear big picture.